abci/client/doc.go

```diff
@@ -0,0 +1,29 @@
+// Package client provides an ABCI implementation in Go.
+//
+// There are 3 clients available:
+//		1. socket (unix or TCP)
+//		2. local (in memory)
+//		3. gRPC
+//
+// ## Socket client
+//
+// async: the client maintains an internal buffer of a fixed size. when the
+// buffer becomes full, all Async calls will return an error immediately.
```

Comment:
Maybe this should be the responsibility of the caller, by using e.g. a waitgroup or semaphore to bound the number of calls? That way the caller can decide what should happen when the queue is full, e.g. block, drop, or error, and different callers can control their own limits independently of each other.

Reply:
I feel like this would complicate things unnecessary.